diff --git a/dist/react-tooltip.d.ts b/dist/react-tooltip.d.ts
index 7d75971d355b6998201bab5b0b661f5064f0af43..93c1bd5cfbe609a52fdf036bfaf94f7f7afdabd9 100644
--- a/dist/react-tooltip.d.ts
+++ b/dist/react-tooltip.d.ts
@@ -96,7 +96,8 @@ interface ITooltipWrapper {
 
 declare const TooltipController: ({ id, anchorId, content, html, className, classNameArrow, variant, place, offset, wrapper, children, events, positionStrategy, middlewares, delayShow, delayHide, float, noArrow, clickable, closeOnEsc, style, position, isOpen, setIsOpen, afterShow, afterHide, }: ITooltipController) => JSX.Element;
 
-declare const TooltipProvider: React.FC<PropsWithChildren>;
+// fix applied in https://github.com/ReactTooltip/react-tooltip/pull/940
+declare const TooltipProvider: React.FC<PropsWithChildren<{}>>;
 
 declare const TooltipWrapper: ({ tooltipId, children, className, place, content, html, variant, offset, wrapper, events, positionStrategy, delayShow, delayHide, }: ITooltipWrapper) => JSX.Element;
 
