diff --git a/node_modules/react-terminal-component/lib/react-terminal-component.js b/node_modules/react-terminal-component/lib/react-terminal-component.js
index f568bf0..42c52bd 100644
--- a/node_modules/react-terminal-component/lib/react-terminal-component.js
+++ b/node_modules/react-terminal-component/lib/react-terminal-component.js
@@ -41142,7 +41142,7 @@ function (_Component) {
       }
     });
 
-    _this.emulator = new javascript_terminal__WEBPACK_IMPORTED_MODULE_3__["Emulator"]();
+    _this.emulator = _ref.altEmulator || new javascript_terminal__WEBPACK_IMPORTED_MODULE_3__["Emulator"]();
     _this.historyKeyboardPlugin = new javascript_terminal__WEBPACK_IMPORTED_MODULE_3__["HistoryKeyboardPlugin"](_emulatorState);
     _this.plugins = [_this.historyKeyboardPlugin];
     _this.inputRef = null;
